demo/src/faber_inquirer.ts

```diff
@@ -0,0 +1,133 @@
+import { clear } from 'console'
+import { textSync } from 'figlet'
+import inquirer from 'inquirer'
+
+import { BaseInquirer } from './base_inquirer'
+import { Faber } from './faber'
+import { Listener } from './listener'
+import { Title } from './output_class'
+
+export const runFaber = async () => {
+  clear()
+  console.log(textSync('Faber', { horizontalLayout: 'full' }))
+  const faber = await FaberInquirer.build()
+  faber.processAnswer()
+}
+
+enum PromptOptions {
+  Connection = 'setup connection',
+  Credential = 'offer credential',
+  Proof = 'request proof',
+  Message = 'send message',
+  Exit = 'exit',
+  Restart = 'restart',
+}
+
+export class FaberInquirer extends BaseInquirer {
+  public faber: Faber
+  public promptOptionsString: string[]
+  public listener: Listener
+
+  public constructor(faber: Faber) {
+    super()
+    this.faber = faber
+    this.listener = new Listener()
+    this.promptOptionsString = Object.values(PromptOptions)
+    this.listener.messageListener(this.faber.agent, this.faber.name)
+  }
+
+  public static async build(): Promise<FaberInquirer> {
+    const faber = await Faber.build()
+    return new FaberInquirer(faber)
+  }
+
+  private async getPromptChoice() {
+    const prompt = inquirer.prompt([this.inquireOptions(this.promptOptionsString)])
+    return prompt
+  }
+
+  public async processAnswer() {
+    const choice = await this.getPromptChoice()
+    if (this.listener.on === true) {
+      return
+    }
+    switch (choice.options) {
+      case PromptOptions.Connection:
+        await this.connection()
+        break
+      case PromptOptions.Credential:
+        await this.credential()
+        return
+      case PromptOptions.Proof:
+        await this.proof()
+        return
+      case PromptOptions.Message:
+        await this.message()
+        break
```

Comment:
are all these return/break correct? Looks a bit random, but I might be wrong

Reply:
Yeah because with the Credential and Proof use case, you get a new prompt afterwards (not the main menu), once that prompt is finished, you get the main menu but this is called from the other prompt.. I'll look into if it's possible to do this differently!